File: client/Utils/NotificationModel.swift


import Foundation

// MARK: - NotificationModel
struct NotificationModel: Codable {
    let notification: [NotificationViewModel]
}

// MARK: - Notification
struct NotificationViewModel: Codable {
    let id, alertName, alertID, actualDatetime: String
    let priority, notificationSentDatetime, notificationMessage: String?
    let v: Int

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case alertName
        case alertID = "alertId"
        case actualDatetime, priority, notificationSentDatetime, notificationMessage
        case v = "__v"
    }
}

File: Backend/models/notification.model.js
const mongoose = require("mongoose");
const NOTIFICATION = mongoose.model(
  "notifications",
  new mongoose.Schema({
    alertName: String,
    alertId: String,
    actualDatetime:String,
    sourceIP: String,
    destinationIP: String,
    protocol:String,
    priority:String,
    attackLevel:String,
    notificationMessage: String,
    notificationId:String,
    isSent:Boolean,
    notificationSentDatetime: String,

  })
);
module.exports = NOTIFICATION;